File: solutions/fsharp/booking-up-for-beauty/BookingUpForBeauty.fs
module BookingUpForBeauty

open System

let schedule (appointmentDateDescription: string) : DateTime =
    DateTime.Parse(appointmentDateDescription)

let hasPassed (appointmentDate: DateTime) : bool =
    appointmentDate <= DateTime.Now

let isAfternoonAppointment (appointmentDate: DateTime) : bool =
    let h = appointmentDate.Hour
    h >= 12 && h < 18

let description (appointmentDate: DateTime) : string =
    $"You have an appointment on {appointmentDate.ToString()}."

let anniversaryDate () : DateTime = DateTime(DateTime.Now.Year, 9, 15)

File: solutions/julia/booking-up-for-beauty/booking-up-for-beauty.jl
import Dates

input_dateformat = dateformat"m/d/y H:M:S"
output_dateformat = dateformat"E, U d, Y at HH:MM"

function schedule_appointment(appointment::String)
    Dates.DateTime(appointment, input_dateformat)
end

function has_passed(appointment::DateTime)
    appointment < Dates.now()
end

function is_afternoon_appointment(appointment::DateTime)
    12 <= Dates.hour(appointment) < 18
end

function describe(appointment::DateTime)
    "You have an appointment on $(Dates.format(appointment, output_dateformat))"
end

function anniversary_date()
    Dates.Date(Dates.year(Dates.now()), 9, 15)
end
